src/amo/components/LandingPage.js

```diff
@@ -41,9 +44,33 @@ export class LandingPageBase extends React.Component {
 
   static defaultProps = {
     apiAddonType: getApiAddonType,
+    contentForType: null,
   }
 
-  contentForType(visibleAddonType) {
+  componentWillMount() {
+    this.setViewContextType();
+  }
+
+  componentDidUpdate() {
+    this.setViewContextType();
+  }
+
+  setViewContextType() {
+    const { apiAddonType, dispatch, params } = this.props;
+
+    try {
+      const addonType = apiAddonType(params.visibleAddonType);
+      dispatch(setViewContext(addonType));
+    } catch (err) {
+      if (err instanceof AddonTypeNotFound) {
+        log.info('AddonTypeNotFound in setViewContextType()', err);
```

Comment:
Okay, I understand what's happening now. There is a legit 404 case that we need to handle. Thanks for taking the time to explain it in detail.

Here's what I propose we do. Let me know what you think.

1. revert your last commit so that the error gets caught, logged, and ignored in `setViewContextType()` and the `<NotFound/>` component gets rendered as before.
2. add a comment to the catch/ignore code in `setViewContextType()` explaining that the error is handled more properly in `render()`
3. add a `TODO` comment in `render()` above the `<NotFound/>` code saying that it should be removed after https://github.com/mozilla/addons-frontend/issues/2161